lib/manager/mix/extract.js

```diff
@@ -0,0 +1,64 @@
+const { isValid } = require('../../versioning/hex');
+
+module.exports = {
+  extractPackageFile,
+};
+
+function extractPackageFile(content) {
+  logger.trace('mix.extractPackageFile()');
+  const deps = [];
+  const contentArr = content.split('\n');
+  const depSectionRegExp = /defp\s+deps.*do/g;
+  const depMatchRegExp = /{:(\w+),\s*([^:"]+)?:?\s*"([^"]+)",?\s*(organization: "(.*)")?.*}/gm;
+
+  for (let lineNumber = 0; lineNumber < contentArr.length; lineNumber += 1) {
+    if (contentArr[lineNumber].match(depSectionRegExp)) {
+      logger.trace(`Matched dep section on line ${lineNumber}`);
+      let depBuffer = '';
+      do {
+        depBuffer += contentArr[lineNumber] + '\n';
+        lineNumber += 1;
+      } while (!contentArr[lineNumber].includes('end'));
+      let depMatch;
+      do {
+        depMatch = depMatchRegExp.exec(depBuffer);
+        if (depMatch) {
+          const depName = depMatch[1];
+          const datasource = depMatch[2];
+          const currentValue = depMatch[3];
+          const organization = depMatch[5];
+
+          const dep = {
+            depName,
+            currentValue,
+          };
+
+          if (datasource) {
+            dep.datasource = datasource;
+            dep.depType = datasource;
```

Comment:
What is an example of using a non-hex datasource?

Reply:
I don't think this depType is right. If there is no distinction between deps/dev deps/etc then there's no reason to have a depType. Otherwise, it should be something meaningful.

Reply:
If there are no "sections" or types of dependencies then let's leave depType to null.